src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -269,18 +268,16 @@ class BasePaymentsPage extends React.Component {
                 {Permissions.canUseWallet(this.props.betas) && (
                     <>
                         <View style={[styles.mv5]}>
-                            {this.props.isLoadingPaymentMethods && !this.props.network.isOffline ? (
-                                <ActivityIndicator color={themeColors.spinner} size="large" />
-                            ) : (
-                                <OfflineWithFeedback
-                                    pendingAction={CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD}
-                                    errors={this.props.walletTerms.errors}
-                                    onClose={PaymentMethods.clearWalletTermsError}
-                                    errorRowStyles={[styles.ml10, styles.mr2]}
-                                >
-                                    <CurrentWalletBalance />
-                                </OfflineWithFeedback>
-                            )}
+                            <OfflineWithFeedback
+                                pendingAction={CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD}
+                                errors={this.props.walletTerms.errors}
+                                onClose={PaymentMethods.clearWalletTermsError}
+                                errorRowStyles={[styles.ml10, styles.mr2]}
+                            >
+                                <CurrentWalletBalance
+                                    showActivityIndicator={this.props.isLoadingPaymentMethods && !this.props.network.isOffline}
```

Comment:
It seems odd to have this prop passed down that controls something that could be entirely internal to the component. In fact, this is the only place that we use `isLoadingPaymentMethods` in this component, so I think we should have CurrentWalletBalance handle this functionality itself.

Reply:
Yeah, but now that I'm looking at it it seems weird that this small component taps into `isLoadingPaymentMethods`. Why would a current wallet balance care about whether payment methods are loading.

BasePaymentPage loads the wallet amount and payment methods. And yet, the currentWalletBalance reaches out of its scope to check on whether Payment methods have loaded yet. 

CurrentWalletBalance is also used when you click "Transfer". So it would be weird if it has UI related to payment method loading is in the component.